xikolo-ios/ViewControllers/Courses/CourseListViewController.swift

```diff
@@ -31,6 +33,27 @@ class CourseListViewController : AbstractCourseListViewController {
         self.collectionView?.emptyDataSetDelegate = nil
     }
 
+    lazy var refreshControl: UIRefreshControl = {
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action:
+            #selector(CourseListViewController.handleRefresh(_:)),
+                                 for: UIControlEvents.valueChanged)
+        return refreshControl
+    }()
+    
+    func handleRefresh(_ refreshControl: UIRefreshControl) {
+        if UserProfileHelper.isLoggedIn() {
+            EnrollmentHelper.syncEnrollments()
+        }else{
+            CourseHelper.refreshCourses()
+        }
+        endRefreshing()
```

Comment:
Right. You should make use of the Future returned by `EnrollmentHelper.syncEnrollments()` resp. `CourseHelper.refreshCourses()`